docs/mainnet-examples/OptimismMultisig.md

```diff
@@ -0,0 +1,132 @@
+        );
+        vm.makePersistent(address(addresses));
+
+        super.run();
+    }
+
+    function deploy() public override {
+        if (!addresses.isAddressSet("OPTIMISM_L1_NFT_BRIDGE_IMPLEMENTATION")) {
+            address l1NFTBridgeImplementation = address(new MockUpgrade());
+
+            addresses.addAddress(
+                "OPTIMISM_L1_NFT_BRIDGE_IMPLEMENTATION",
+                l1NFTBridgeImplementation,
+                true
+            );
+        }
+    }
+
+    function build()
+        public
+        override
+        buildModifier(addresses.getAddress("OPTIMISM_MULTISIG"))
+    {
+        IProxyAdmin proxy = IProxyAdmin(
+            addresses.getAddress("OPTIMISM_PROXY_ADMIN")
+        );
+
+        proxy.upgrade(
+            addresses.getAddress("OPTIMISM_L1_NFT_BRIDGE_PROXY"),
+            addresses.getAddress("OPTIMISM_L1_NFT_BRIDGE_IMPLEMENTATION")
+        );
+    }
+
+    function simulate() public override {
+        address multisig = addresses.getAddress("OPTIMISM_MULTISIG");
+
+        _simulateActions(multisig);
+    }
+
+    function validate() public override {
+        IProxy proxy = IProxy(
+            addresses.getAddress("OPTIMISM_L1_NFT_BRIDGE_PROXY")
+        );
+
+        // implementation() caller must be the owner
+        vm.startPrank(addresses.getAddress("OPTIMISM_PROXY_ADMIN"));
+        require(
+            proxy.implementation() ==
+                addresses.getAddress("OPTIMISM_L1_NFT_BRIDGE_IMPLEMENTATION"),
+            "Proxy implementation not set"
+        );
+        vm.stopPrank();
+    }
+}
+```
+
+Let's go through each of the functions that are overridden.
+
+-   `name()`: Define the name of your proposal.
+-   `description()`: Provide a detailed description of your proposal.
+-   `deploy()`: This example demonstrates the deployment of new MockUpgrade which will be
+    used as new implementation to proxy.
+-   `run()`: Sets environment for running the proposal. It sets `addresses`. `addresses` is address object
+    containing addresses to be used in proposal that are fetched from `Addresses.json`.
+-   `build()`: In this example, L1 NFT Bridge is upgraded to new implementation. The actions
+    should be written in solidity code and in the order they should be executed. Any calls (except to the Addresses object) will be recorded and stored as actions to execute in the run function. `caller` address is passed into `buildModifier` that will call actions in `build`. Caller is optimism multisig for this example.
+-   `simulate()`: Execute the proposal actions outlined in the `build()` step. This
+    function performs a call to `_simulateActions()` from the inherited
+    `MultisigProposal` contract. Internally, `_simulateActions()` simulates a call to the [Multicall3](https://www.multicall3.com/) contract with the calldata generated from the actions set up in the build step.
+-   `validate()`: It validates implementation is upgraded correctly.
```

Comment:
> There are a lot of places where these steps are essentially repeated, so these should be consolidated in places where they can for brevity.

These are already consolidated here [proposal-functions.md](https://github.com/solidity-labs-io/forge-proposal-simulator/blob/feat/docs-update/docs/overview/architecture/proposal-functions.md)

> This content should be differentiated, and we shouldn't be repeating ourselves so much. There should be a single place where each step of the proposal is explained, and then the individual docs talk about how this proposal's version of that function implementation differs or works.

I think this repetition is needed because a user might go through only a single example and it will be easier for that user to walk through that example if we repeat these details.

> For the examples, we should show Solidity code snippets at each step to demonstrate to developers how this works under the hood, and draw out the state machines so they can see how each step does or does not mutate state. All assumptions should be explicitly stated, nothing should be left to chance.

Agree with this. Making this update.